internal/report/snapshot/snapshot_buffers.go

```diff
@@ -86,11 +86,15 @@ func (l *LogBuffer) Add(ld plog.Logs) {
 	}
 }
 
+// LogsMarshaller marshals logs to a byte representation
+type LogsMarshaller interface {
+	MarshalLogs(ld plog.Logs) ([]byte, error)
+}
+
 // ConstructPayload condenses the buffer and serializes to protobuf
-func (l *LogBuffer) ConstructPayload(searchQuery *string, minimumTimestamp *time.Time) ([]byte, error) {
+func (l *LogBuffer) ConstructPayload(logsMarshaler LogsMarshaller, searchQuery *string, minimumTimestamp *time.Time) ([]byte, error) {
```

Comment:
This change allows the caller to decide how the buffer is marshalled (in this case, it's so the original method for snapshots can use protobuf marshalling, but the new stuff can use json marshalling)